### ISSUE TITLE
Parallel route with missing loading.tsx leads to blank page on navigation

### ISSUE THREAD

**Issue by Kaeldehta** [vercel/next.js#56701]
### Link to the code that reproduces this issue

https://github.com/Kaeldehta/next-parallel-routes-bug

### To Reproduce

1. Build production build using `npm run build`
2. Start server using `npm run start`
3. Navigate to any random page except root page 
4. Click the navigate link
5. Page goes blank

### Current vs. Expected behavior

Next should navigate to the correct page while showing streaming UI only for the page, but not for the parallel route. 
Or it should give an error that a loading file is missing.
Instead the whole site goes blank without any error.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:01:02 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8112
Binaries:
  Node: 18.17.0
  npm: 9.8.1
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 13.5.5-canary.5
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Additional context

This happens only in production builds, also on Vercel.
In the development server it works without problems.

**Comment by wscotten** [comment]
bump

**Comment by ztanner** [comment]
Hi there! Thanks for the report. I believe this has been recently fixed by https://github.com/vercel/next.js/pull/61597 -- I tested your repro against the latest canary and it appears to be behaving as expected. I'm going to close this out but let us know in case you're still running into any trouble.

**Comment by mrjasonroy** [comment]
Confirmed fix in 14.1.1-canary.52, as related #61597

Example repo with broken and working branches:

https://github.com/mrjasonroy/nextjs-parallel-route-14.1.0-errors/tree/14.1.1-canary-52
